crates/bevy_ecs/src/bundle.rs

```diff
@@ -1230,7 +1365,7 @@ fn initialize_dynamic_bundle(
     let id = BundleId(bundle_infos.len());
     let bundle_info =
         // SAFETY: `component_ids` are valid as they were just checked
-        unsafe { BundleInfo::new("<dynamic bundle>", components, component_ids, id) };
+        unsafe { BundleInfo::new("<dynamic bundle>", components, component_ids, Vec::new(), id) };
```

Comment:
This `Vec::new()` seems wrong, as it means that dynamically inserting a component won't insert its required components.

Reply:
And here's a test that fails due to this:

```rs
#[derive(Component)]
#[require(Y)]
struct X;

#[derive(Component, Default)]
struct Y;

let mut world = World::new();
let x_id = world.init_component::<X>();

let mut e = world.spawn_empty();

bevy_ptr::OwningPtr::make(X, |ptr| unsafe {
    e.insert_by_id(x_id, ptr);
});

assert!(e.contains::<Y>());
```

Reply:
I've reworked the dataflow a bit to make this possible (RequiredComponents are now stored on ComponentInfo). Thanks for the test. It now passes!